tests/language-feature/struct-field-initializers/struct-field-initializer-extension.slang

```diff
@@ -0,0 +1,56 @@
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain 
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -emit-spirv-directly
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-cpu -compute -entry computeMain
+//TEST(smoke,compute):COMPARE_COMPUTE(filecheck-buffer=BUF):-dx12 -use-dxil -compute -entry computeMain
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<int> outputBuffer;
+
+struct DefaultStructNoInit
+{
+    int data0 = 2;
+    int data1 = 2;
+
+};
+extension DefaultStructNoInit
+{
+    __init(int indata)
+    {
+        //unused
+    }
+}
+
+struct DefaultStructWithInit
+{
+    int data0;
+    int data1 = 3;
+    int data2;
+};
+extension DefaultStructWithInit
+{
+    __init(int indata)
+    {
+        //unused
+    }
+    __init()
+    {
+        data0 = 3;
+        data2 = 3;
+    }
+}
+
+[numthreads(1, 1, 1)]
+void computeMain(uint3 dispatchThreadID: SV_DispatchThreadID)
+{
+    DefaultStructNoInit noInit = DefaultStructNoInit();
+    DefaultStructWithInit withInit = DefaultStructWithInit();
```

Comment:
I am little worried that there is no testing for the constructor that takes an argument.
Maybe you can modify this test as following,
```
extension DefaultStructWithInit
{
    __init(int indata)
    {
        data0 = indata; // <== Changes here
        data2 = 0; // <== Changes here
    }
    __init()
    {
        data0 = 3;
        data2 = 3;
    }
}

[numthreads(1, 1, 1)]
void computeMain(uint3 dispatchThreadID: SV_DispatchThreadID)
{
    DefaultStructNoInit noInit = DefaultStructNoInit();
    DefaultStructWithInit withInit0 = DefaultStructWithInit();
    DefaultStructWithInit withInit1 = DefaultStructWithInit(4); // <== Changes here
```

Reply:
This test, tests a failure of using default ctor as an extension and to be sure Slang does not crash (for language server). The goal is to fail if using a default ctor in `extension`.

I will add a separate test for parameter functions in `extension`s which are designed to run. I will also add a test for constructor with parameter to ensure it also works as intended.